mmtrack/datasets/trackingnet_dataset.py

```diff
@@ -1,21 +1,118 @@
 # Copyright (c) OpenMMLab. All rights reserved.
+import glob
 import os
 import os.path as osp
 import shutil
-from collections import defaultdict
+import time
 
+import numpy as np
 from mmdet.datasets import DATASETS
 
-from .sot_test_dataset import SOTTestDataset
+from .base_sot_dataset import BaseSOTDataset
 
 
 @DATASETS.register_module()
-class TrackingNetDataset(SOTTestDataset):
-    """TrackingNet dataset for the testing of single object tracking.
+class TrackingNetDataset(BaseSOTDataset):
+    """TrackingNet dataset of single object tracking.
 
-    The dataset doesn't support training mode.
+    The dataset can both support training and testing mode.
     """
 
+    def __init__(self, num_chunks=12, *args, **kwargs):
+        """Initialization of SOT dataset class.
+
+        Args:
+            num_chunks (int, optional): the number of chunks. Some methods may
+                only use part of the dataset. Default to all chunks, 12.
+        """
+        self.num_chunks = num_chunks
```

Comment:
The `num_chunks` should not be a `int`, since the user may want to specify the index of chunks. It should be a list with optional values in `[1,2,3,4,'all']`